tests/test_kernels.py

```diff
@@ -104,111 +122,42 @@ def test_required_grid_vars(grid_type_field_and_extra_kwargs):
 
 # The following definition of irregular_grids is hard coded; maybe a better definition
 # would be: all grids that have len(required_grid_vars)>1 (more than just a wet_mask)
-irregular_grids = [
-    gt for gt in GridType if "IRREGULAR" in gt.name or "TRIPOLAR_POP" in gt.name
-]
+irregular_grids = [GridType.IRREGULAR_WITH_LAND, GridType.TRIPOLAR_POP_WITH_LAND]
 
 
-def test_flux_in_y_direction(grid_type_field_and_extra_kwargs):
+def test_flux(grid_type_field_and_extra_kwargs):
     """This test checks that the Laplacian computes the correct fluxes in y-direction if the grid is irregular.
     The test will catch sign errors in the Laplacian rolling of array elements along the y-axis."""
     grid_type, data, extra_kwargs = grid_type_field_and_extra_kwargs
 
-    if grid_type in irregular_grids:
-        # deploy mass at random location away from Antarctica: delta_{j,i}
-        delta = np.zeros_like(data)
-        ny = np.shape(delta)[0]
-        random_yloc = np.random.randint(5, ny - 2)
-        nx = np.shape(delta)[1]
-        random_xloc = np.random.randint(0, nx)
-        delta[random_yloc, random_xloc] = 1
-
-        test_kwargs = copy.deepcopy(extra_kwargs)
-        ones = np.ones_like(data)
-        # create highly irregular grid data for
-        # - southern edge of cell {j-1,i};
-        # - southern edge of cell {j+2,i};
-        # this should leave laplacian(delta_{j,i}) unaffected
-        if grid_type == GridType.IRREGULAR_WITH_LAND:
-            test_kwargs["wet_mask"] = ones  # no land for simplicity
-            test_kwargs["area"] = ones
-            test_kwargs["dyw"] = ones
-            test_kwargs["dys"] = ones
-            test_kwargs["dxw"] = ones
-            test_kwargs["dxs"] = ones.copy()
-            test_kwargs["dxs"][random_yloc - 1, :] = 1000
-            test_kwargs["dxs"][random_yloc + 2, :] = 2000
-        if grid_type == GridType.TRIPOLAR_POP_WITH_LAND:
-            test_kwargs["wet_mask"] = ones.copy()
-            test_kwargs["wet_mask"][0, :] = 0  # Antarctica
-            test_kwargs["tarea"] = ones
-            test_kwargs["dye"] = ones
-            test_kwargs["dyn"] = ones
-            test_kwargs["dxe"] = ones
-            test_kwargs["dxn"] = ones.copy()
-            test_kwargs["dxn"][random_yloc - 2, :] = 1000
-            test_kwargs["dxn"][random_yloc + 1, :] = 2000
```

Comment:
Since the irregular grids are all now automatically being populated with random data, my thinking was that it should be unnecessary to do this sort of manual manipulation of the grid variables to create anisotropy.

Reply:
This test was designed for regular grid data, so I reverted back to an earlier version of yours in #80. For more details, see https://github.com/ocean-eddy-cpt/gcm-filters/pull/80#issue-701806629.